backend/services/core/project_settings_service.py

```
@@ -137,6 +137,72 @@ def get_all_settings(self, project_id: str) -> Dict[str, Any]:
+                    logger.warning(f"Corrupted settings file for project {project_id}, creating new one")
+                    settings = {}
+            
+            # Update the selected LLM model
+            settings['selected_llm_model_id'] = model_id
+            
+            # Write the updated settings
+            with open(settings_file, 'w', encoding='utf-8') as f:
+                json.dump(settings, f, indent=2, ensure_ascii=False)
+            
+            logger.info(f"Successfully set selected LLM model to '{model_id}' for project {project_id}")
+            return True
```

Ruff (0.12.2)

Consider moving this statement to an `else` block

(TRY300)